File: Sample/BasicSample/sample_cs/Browser/SampleBrowser.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

class SampleBrowser
{
    public static readonly float Margin = 16;

    private ISample[] samples;

    public ISample Selected = null;
    private SampleBrowserLayer browserLayer = null;
    private SampleInfoLayer infoLayer = null;

    public SampleBrowser(ISample[] samples)
    {
        this.samples = samples;
    }

    public void ShowInfo(ISample sample)
    {
        infoLayer.Show(sample);
    }

    public void Run()
    {
        while (true)
        {
            asd.Engine.Initialize("サンプルブラウザ", 640, 480, new asd.EngineOption());

            var scene = new asd.Scene();
            browserLayer = new SampleBrowserLayer(this, samples);

            var viewSize = SampleBrowserLayer.Columns * SampleBrowserLayer.ItemOffset.Y;
            var size = (480 - 80) * viewSize / browserLayer.TotalHeight;
            infoLayer = new SampleInfoLayer(size, browserLayer.TotalHeight, viewSize) { DrawingPriority = 2 };

            asd.Engine.ChangeScene(scene);
            scene.AddLayer(browserLayer);
            scene.AddLayer(infoLayer);

            var hintLayer = new asd.Layer2D();
            hintLayer.AddObject(new asd.TextureObject2D()
            {
                Texture = asd.Engine.Graphics.CreateTexture2D("Data/Browser/Hint.png")
            });

            scene.AddLayer(hintLayer);

            while (asd.Engine.DoEvents() && Selected == null)
            {
                asd.Engine.Update();
                infoLayer.MoveScrollBar(browserLayer.CameraArea.Y);
            }

            asd.Engine.Terminate();

            if (Selected == null)
            {
                break;
            }
            Selected.Run();
            Selected = null;
        }
    }
}

File: Sample/ApplicationSample/sample_cpp/Browser/SampleBrowser.cpp
#include "SampleBrowser.h"
#include "SampleBrowserLayer.h"
#include "SampleInfoLayer.h"
#include <memory>
#include <Altseed.h>

using namespace std;

SampleBrowser::SampleBrowser(std::vector<SampleInfo>& samples)
	: m_samples(samples)
{
}

void SampleBrowser::Run()
{
	while (true)
	{
		Selected.isAvailable = false;
		asd::Engine::Initialize(asd::ToAString("サンプルブラウザ").c_str(), 640, 480, asd::EngineOption());

		auto scene = make_shared<asd::Scene>();
		browserLayer = make_shared<SampleBrowserLayer>(this, m_samples);

		auto size = (480 - 85) * (480 - 85) / browserLayer->GetTotalHeight();
		infoLayer = make_shared<SampleInfoLayer>(size, browserLayer->GetTotalHeight());
		infoLayer->SetDrawingPriority(2);

		asd::Engine::ChangeScene(scene);
		scene->AddLayer(browserLayer);
		scene->AddLayer(infoLayer);

		auto hintLayer = make_shared<asd::Layer2D>();
		auto hintObject = make_shared<asd::TextureObject2D>();
		hintObject->SetTexture(asd::Engine::GetGraphics()->CreateTexture2D(asd::ToAString("Data/Browser/Hint.png").c_str()));
		hintLayer->AddObject(hintObject);

		scene->AddLayer(hintLayer);

		while (asd::Engine::DoEvents() && !Selected.isAvailable)
		{
			asd::Engine::Update();
			infoLayer->MoveScrollBar(browserLayer->GetCameraArea().Y);
		}

		asd::Engine::Terminate();

		if (!Selected.isAvailable)
		{
			break;
		}
		Selected.func();
		Selected.isAvailable = false;

	}

}

void SampleBrowser::ShowInfo(SampleInfo sample)
{
	infoLayer->Show(sample);
}